onnxruntime/core/graph/graph.cc

```diff
@@ -1297,6 +1306,7 @@ void Graph::ReverseDFSFrom(const std::vector<const Node*>& from,
     if (comp) {
       std::vector<const Node*> sorted_nodes;
       for (auto iter = n.InputNodesBegin(); iter != n.InputNodesEnd(); ++iter) {
+        if (stop && stop(&n, &(*iter))) continue;
```

Comment:
> if (stop && stop(&n, &(*iter))) continue; [](start = 7, length = 42)

can these two check on stop move to line 1301 and merge to one check? So basically we always push input nodes, but when visiting we'll skip when stop == true.

Reply:
So the stop takes as input the current node (_from_ node) and the proposed neighbor to move to (_to_ node). Once the node is added to the stack, the preceding node it came from is lost. We can add that to the stack, but it seems simpler and more intuitive to me to perform the check beforehand. As a practical matter the number of calls will be the same.

Reply:
I see, thanks for the explanation, let's keep it this way.

---
In reply to: [436057581](https://github.com/microsoft/onnxruntime/pull/4116#discussion_r436057581) [](ancestors = 436057581)